### PR TITLE
Scrape all available build-ids for all releases.

### LINKED ISSUE #110: Build id information should be a required information
Right now most build records in the database don't seem to have any `build` object attached to them, while it should contains the build id and build date, which we declared critical information for this project.

For instance, we currently only seem to have 43 build records exposing that information: https://kinto-ota.dev.mozaws.net/v1/buckets/build-hub/collections/releases/records?_limit=10&_sort=-build.date&source.product=firefox&not_build.id

I think the build objet should be a required information, and/or we shouldn't list builds without that information. A compromise could probably be to have an option in the search UI to allow or disallow listing builds with that information missing.

### PR DESCRIPTION
**Pull request by Natim**
Fixes #110

### CODE REVIEW

File: jobs/buildhub/utils.py [hunk 1]
```diff
@@ -11,6 +11,19 @@
     }
 
 
+def localize_nightly_url(nightly_url):
```
Comment (Natim) [inline jobs/buildhub/utils.py]: It is ugly, I know, but tested... And yes scrapping is ugly I guess.

File: jobs/buildhub/scrape_archives.py [hunk 2]
```diff
@@ -180,16 +181,24 @@ def archive_url(product, version=None, platform=None, locale=None, nightly=None,
...
-    # XXX: It is only available for en-US though. Should we use the same for every locale?
-    if "en-US" not in nightly_url:
+    if not is_release_filename(product, os.path.basename(nightly_url)):
         return None
 
+    # Make sure the nightly_url is turned into a en-US one.
+    nightly_url = localize_nightly_url(nightly_url)
+
+    if nightly_url in _nightly_metadata:
+        return _nightly_metadata[nightly_url]
```
Comment (glasserc) [inline jobs/buildhub/scrape_archives.py]: Is it possible to use a `memoize` decorator on an async function instead of this handwritten one?

File: jobs/buildhub/utils.py (hunk 1 above)
Comment (n1k0) [inline jobs/buildhub/utils.py]: I couldn't agree more (and I wrote a famous scraper).

File: jobs/buildhub/scrape_archives.py [hunk 3]
```diff
@@ -180,16 +181,24 @@ def archive_url(product, version=None, platform=None, locale=None, nightly=None,
...
-async def fetch_nightly_metadata(session, nightly_url):
+_nightly_metadata = {}
+
+async def fetch_nightly_metadata(session, product, nightly_url):
     """A JSON file containing build info is published along the nightly archive.
     """
-    # XXX: It is only available for en-US though. Should we use the same for every locale?
-    if "en-US" not in nightly_url:
+    if not is_release_filename(product, os.path.basename(nightly_url)):
         return None
```
Comment (leplatrem) [inline jobs/buildhub/scrape_archives.py]: Why do we have to do this here? Can you explain in a comment? Why not do this in the calling code?

File: jobs/buildhub/utils.py [hunk 4]
```diff
@@ -11,6 +11,19 @@
...
+    nightly_url = nightly_url.replace('-l10n', '')
+    parts = nightly_url.split('.')
+    locale = 'en-US'
+    index = -3
+    if 'tar' in nightly_url:
+        index = -4
+    if 'mobile' in nightly_url:
+        locale = 'multi'
+    parts[index] = locale
+    return '.'.join(parts)
```
Comment (leplatrem) [inline jobs/buildhub/utils.py]: I usually prefer regexps, but it may be a matter of taste...

Also could you add a comment here to explain what does this function? At least with an example and why we have to do this

File: jobs/buildhub/scrape_archives.py (hunk 2 above)
Comment (Natim) [inline jobs/buildhub/scrape_archives.py]: It should be possible yes. Can we do it in another PR?

File: jobs/buildhub/scrape_archives.py (hunk 3 above)
Comment (Natim) [inline jobs/buildhub/scrape_archives.py]: Yes we can do it in the calling code.

Comment (Natim) [inline jobs/buildhub/scrape_archives.py]: So I did it there because `fetch_nightly_metadata` was called with invalid URLs, it made sense for me to validate it inside the function itself.

File: jobs/buildhub/scrape_archives.py (hunk 2 above)
Comment (Natim) [inline jobs/buildhub/scrape_archives.py]: Refs https://github.com/mozilla-services/buildhub/issues/144
